filters/filter_test.go

```diff
@@ -287,9 +302,25 @@ func TestFilters(t *testing.T) {
 				}
 			}
 
-			if !reflect.DeepEqual(results, testcase.expected) {
+			if compareEvents(results, testcase.expected) {
```

Comment:
`assert.Equal()` is already being used in this repo. Probably makes more sense to use that instead of writing a custom assertion that does the same thing.

Reply:
We need to compare two slice. and The element of slice has different order. So assert.Equal cann't do it.

Reply:
> The element of slice has different order. 

If this is the case, something has been broken. The filtered set should have the same order as the original set. Moving away from reflect.DeepEqual here is a bug in this "refactoring".

Like I said, remove the changes to filter_test.go from this PR.